odoo/orm/domains.py

```diff
@@ -364,6 +367,17 @@ def validate(self, model: BaseModel) -> None:
         # just execute the optimization code that goes through all the fields
         self.optimize(model, full=True)
 
+    def filter_records(self, records: M) -> M:
+        """Filter a recordset using this domain."""
+        if not records:
+            return records
+        predicate = self._filter_records(records)
+        return records.browse(record_id for record_id, record in zip(records._ids, records) if predicate(record))
```

Comment:
Isn't `zip` a bit overkill?

Reply:
Actually, accessing `.id` is quite costly compared to the zip here.
```
In [9]: %timeit records.browse(rec.id for rec in records if True)
1.78 ms ± 25.4 μs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [10]: %timeit records.browse(record_id for record_id, rec in zip(records._ids, records) if True)
1.28 ms ± 6.85 μs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```